arch/arm/src/arm/Toolchain.defs

```diff
@@ -103,21 +103,12 @@ OBJDUMP = $(CROSSDEV)objdump
 
 # Add the builtin library
 
-EXTRA_LIBS += -lgcc
-EXTRA_LIBPATHS += -L "${shell dirname "`$(CC) $(ARCHCPUFLAGS) --print-libgcc-file-name`"}"
+EXTRA_LIBS := ${shell $(CC) $(ARCHCPUFLAGS) --print-file-name=libgcc.a}
```

Comment:
what's the motivation to change --print-libgcc-file-name to --print-file-name=libgcc.a?

iirc, the former has a better chance to work for clang configured to use non-libgcc runtime. (eg. compiler_rt)
i don't know if it matters here though.

Reply:
I thought both are equal. Let me provide a patch fix it.

Reply:
FYI, https://reviews.llvm.org/D25338
as far as i know, the situation has not been changed since then.

> Maybe, it's better to add clang to the ci build.

i agree.

Reply:
Fixed here: https://github.com/apache/incubator-nuttx/pull/5385